File: SAMPLES/PYTHON/strings.py
# strings.py - Exemple PYTHON86 : operations sur chaines, find, upper, split
# Compile : PYTHON86.EXE SAMPLES\PYTHON\strings.py strings.asm

# Concatenation et repetition
print("=== Concatenation et repetition ===")
s1 = "Hello"
s2 = " World"
s3 = s1 + s2
print("concat :", s3)

sep = "-" * 30
print(sep)

# Longueur
print()
print("=== Longueur ===")
msg = "Bonjour"
print("len(Bonjour) =", len(msg))

# Indexation
print()
print("=== Indexation ===")
abc = "ABCDEF"
print("abc[0] =", abc[0])
print("abc[2] =", abc[2])
print("abc[5] =", abc[5])

# Slicing
print()
print("=== Slicing ===")
texte = "Hello World"
print("texte[0:5] =", texte[0:5])
print("texte[6:11] =", texte[6:11])

# Upper et lower
print()
print("=== Upper / Lower ===")
s = "Hello World"
print("upper :", s.upper())
print("lower :", s.lower())

# Strip
print()
print("=== Strip ===")
s = "  espaces  "
print("strip :", s.strip())

# Find
print()
print("=== Find ===")
s = "Hello World"
pos = s.find("World")
print("find(World) =", pos)
pos = s.find("xyz")
print("find(xyz) =", pos)

# Startswith / endswith
print()
print("=== Startswith / Endswith ===")
s = "Hello World"
if s.startswith("Hello"):
    print("Commence par Hello")
if s.endswith("World"):
    print("Termine par World")

# Replace
print()
print("=== Replace ===")
s = "Hello World"
s2 = s.replace("World", "Python")
print("replace :", s2)

# Operateur in
print()
print("=== Operateur in ===")
s = "Hello World"
if "World" in s:
    print("World trouve dans la chaine")
if "xyz" in s:
    print("xyz trouve")
else:
    print("xyz non trouve")

print()
print("Fin de strings.py")

File: SAMPLES/CLOJURE/strings.clj
; strings.clj - str, subs, upper-case, lower-case, trim

; --- str : concatenation ---
(println (str "Hello" " " "World"))
(println (str "abc" 42))
(println (str))

; --- subs : sous-chaine ---
(println (subs "Hello World" 6))
(println (subs "Hello World" 0 5))

; --- count : longueur de chaine ---
(println (count "Hello"))
(println (count ""))

; --- get : caractere par index ---
(println (get "abcde" 0))
(println (get "abcde" 4))

; --- clojure.string/upper-case ---
(println (clojure.string/upper-case "hello world"))

; --- clojure.string/lower-case ---
(println (clojure.string/lower-case "HELLO WORLD"))

; --- clojure.string/trim ---
(println (clojure.string/trim "  espaces  "))

; --- clojure.string/includes? ---
(println (clojure.string/includes? "hello world" "world"))
(println (clojure.string/includes? "hello" "xyz"))

; --- clojure.string/replace ---
(println (clojure.string/replace "hello world" "world" "clojure"))
(println (clojure.string/replace "aaa" "a" "b"))

; --- clojure.string/join ---
(def items (list "a" "b" "c"))
(println (clojure.string/join ", " items))
(println (clojure.string/join "-" (list "2026" "03" "10")))

; --- str avec types mixtes ---
(println (str "val=" 42 " ok=" true))
